xlators/storage/posix/src/posix-entry-ops.c

```diff
@@ -215,6 +220,32 @@ posix_lookup(call_frame_t *frame, xlator_t *this, loc_t *loc, dict_t *xdata)
     op_ret = -1;
     if (gf_uuid_is_null(loc->pargfid) || (loc->name == NULL)) {
         /* nameless lookup */
+        op_ret = op_errno = 0;
+        /* The gfid will be renamed to ".glusterfs/unlink" in case
+         * there are any open fds on the file in posix_unlink path.
+         * So client can request server to do nameless lookup with
+         * xdata = GF_UNLINKED_LOOKUP in ".glusterfs/unlink"
+         * dir if a client wants to know the status of the all open fds
+         * on the unlinked file. If the file still present in the
+         * ".glusterfs/unlink" dir then it indicates there still
+         * open fds present on the file and the file is still under
+         * unlink process */
+        ret = dict_get_uint32(xdata, GF_UNLINKED_LOOKUP, &lookup_unlink_dir);
+        if (!ret && lookup_unlink_dir) {
+            POSIX_GET_FILE_UNLINK_PATH(priv->base_path, loc->gfid, unlink_path);
+            ret = sys_lstat(unlink_path, &lstatbuf);
+            if (ret) {
+                op_ret = -1;
+                op_errno = ret;
+                if (ret == ENOENT)
+                    buf.ia_nlink = 0;
+            } else {
+                iatt_from_stat(&buf, &lstatbuf);
```

Comment:
We need to set `buf.ia_nlink` to 0 here. The unlinked file exists, but there isn't any name referencing it, so the number of hardlinks should be set to 0.